Tehtris/tehtris_modules/trigger_tehtris_events.py

```diff
@@ -31,20 +32,37 @@ class TehtrisEventConnector(Connector):
 
     def __init__(self, *args, **kwargs):
         super().__init__(*args, **kwargs)
-        self.from_date = datetime.now(timezone.utc) - timedelta(minutes=1)
+        self.context = PersistentJSON("context.json", self._data_path)
+        self.from_date = self.most_recent_date_seen
         self.fetch_events_limit = 100
         self.events_cache: Cache = LRUCache(maxsize=1000)  # TODO: is it enough to have 1000 event ids in cache?
 
+    @property
+    def most_recent_date_seen(self):
+        now = datetime.now(timezone.utc)
+
+        with self.context as cache:
+            most_recent_date_seen_str = cache.get("most_recent_date_seen")
+
+        # if undefined, retrieve events from the last minute
+        if most_recent_date_seen_str is None:
+            return now - timedelta(minutes=1)
+
+        # parse the most recent date seen
+        most_recent_date_seen = isoparse(most_recent_date_seen_str)
+
+        # we don't retrieve messages older than one hour
+        one_hour_ago = now - timedelta(hours=1)
+        if most_recent_date_seen < one_hour_ago:
+            most_recent_date_seen = one_hour_ago
+
+        return most_recent_date_seen
```

Comment:
Cannot we use the [`CheckpointDatetime`](https://github.com/SEKOIA-IO/sekoia-automation-sdk/blob/main/sekoia_automation/checkpoint.py#L141-L146) from the sdk?